src/main/java/com/joyent/manta/monitor/commands/PutFileCommand.java

```diff
@@ -44,13 +45,16 @@ public boolean execute(final MantaOperationContext context) throws Exception {
 
         try {
             stopwatch.start();
+            Histogram.Timer timer = context.getRequestPutHistogramsKey()
+                    .get(context.getChainClassNameKey()).startTimer();
             final MantaHttpHeaders headers = buildHeaders();
             final MantaMetadata metadata = buildMetadata(context);
 
             /* Record the latencies per PUT operation so that we can act upon
              * pathological latency numbers. */
             final MantaObjectResponse response = client.put(
                     filePath, file, headers, metadata);
+            timer.observeDuration();
```

Comment:
I've been giving this PR some thought and what I think is more valuable to measure than just the time to PUT a file is the total time required to PUT directories AND the file. This is the key metric missing from the server-side metrics. This metric tells us effectively how long it actually takes a customer to add a file.

Reply:
This should be pretty straight. Using the 
```
context.getRequestPutHistogramsMap()
                .get(context.getChainClassNameKey()).startTimer();
client.putDirectory(dir, true);
timer.observeDuration();
```
in the ```MkdirCommand.java``` , we can add to the the latency to same histogram object that is being used the ```PutFileCommand.java``` and ```MultipartPutFileCommand.java```, since the metric name will be the same.
The output numbers are such:
```
# HELP manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain Put Request latency in seconds
# TYPE manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain histogram
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.005",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.01",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.025",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.05",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.075",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.1",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.25",} 0.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.5",} 12.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="0.75",} 20.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="1.0",} 21.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="2.5",} 22.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="5.0",} 22.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="7.5",} 30.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="10.0",} 43.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_bucket{le="+Inf",} 44.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_count 44.0
manta_monitor_put_request_latency_seconds_FileUploadGetDeleteChain_sum 190.15839587199997
# HELP manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain Put Request latency in seconds
# TYPE manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain histogram
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.005",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.01",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.025",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.05",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.075",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.1",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.25",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.5",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="0.75",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="1.0",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="2.5",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="5.0",} 0.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="7.5",} 4.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="10.0",} 8.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_bucket{le="+Inf",} 13.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_count 13.0
manta_monitor_put_request_latency_seconds_FileMultipartUploadGetDeleteChain_sum 165.09469757699998
```